### PR TITLE
[REFAPP-52] feat: Skylark GraphQL ingestor implementation

### PR DESCRIPTION
**Pull request by james-wallis**
* [x] Bugfix
* [x] Feature
* [x] Supporting material
* [x] Code style update (formatting, renaming)
* [x] Refactoring (no functional changes, no API changes)
* [x] Infrastructure
* [x] Build related changes
* [x] Documentation content changes
* [x] Other (please describe): < description >

#### Jira

https://ostmodern.atlassian.net/browse/< ticket_id >

#### Self-review
* [ ] Changes self-reviewed

#### Practical Tests Performed (Author)
* < Fill this in or "None" >

#### Practical Tests Performed (Reviewer)
* < Fill this in or "None" >

#### Automated Testing 
* [ ] Appropriate unit/regression/integration tests
* [ ] Tests for negative cases

#### Documentation
* [ ] The PR title is in the format `[< csv_jira_ids >] < descriptive_title >`
* [ ] Anything complex has docstrings/comments explaining the _"why"_
* [ ] System complexity documented on Confluence (only for complex topics if appropriate)

#### Standards & Practices
* [x] [`pre-commit` tooling](https://github.com/ostmodern/ost-githooks) executed (if available)

#### Data Migrations 
* [ ] No database locking migrations are included
* [ ] Database downgrades are present for all upgrades and have been demonstrated (if appropriate/complex/feasible)
* [ ] Old code is forward-compatible with all data migrations

#### Backwards Compatibility 

* [ ] This change includes backwards incompatible changes.

#### Post-release 
* [ ] Follow-up reminders for any post-release configs/actions created

### CODE REVIEW

File: .github/actions/run-ingestor/action.yml [hunk 1]
```diff
@@ -34,6 +34,16 @@ inputs:
   create-additional-objects:
     description: Whether to create additional objects (sets and dynamic objects) which StreamTV requires to run
     required: true
+  saas-ingest:
+    description: Whether to target SaaS Skylark or not
+    required: false
+    default: "false"
```
Comment (freixedelo) [inline .github/actions/run-ingestor/action.yml:40]: we use` ""` in deafult ?

File: packages/ingestor/src/additional-objects/sets.ts [hunk 2]
```diff
@@ -2,13 +2,17 @@
  * Specs for Sets that are created by the ingestor outside of Airtable
  * (Usually you'd create these through the CMS)
  */
-import { SetConfig } from "../interfaces";
+import { SetConfig } from "../lib/interfaces";
 import { quentinTarantinoMovies } from "./dynamicObjects";
 
 const createDataSourceId = (id: string) => `ingestor-set-${id}`;
 
+// Data source but for V10 - GraphQL doesn't like "-"
```
Comment (freixedelo) [inline packages/ingestor/src/additional-objects/sets.ts:10]: what ?! :)

File: packages/ingestor/src/lib/skylark/saas/create.ts [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+        },
+      };
+
+      const graphQLMutation = jsonToGraphQLQuery(mutation);
+
+      const data = await graphQLClient.request<{
+        [key: string]: GraphQLBaseObject;
+      }>(graphQLMutation);
+
+      const arr = values(data);
```
Comment (freixedelo) [inline packages/ingestor/src/lib/skylark/saas/create.ts:51]: this `const` is needed? can we return `values(data)` ?

File: packages/ingestor/src/lib/skylark/saas/create.ts [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+      return arr;
+    })
+  );
+
+  const allData = flatten(chunkedData);
```
Comment (freixedelo) [inline packages/ingestor/src/lib/skylark/saas/create.ts:56]: same as above

File: packages/ingestor/src/lib/skylark/saas/get.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          .filter((recordId) => isNull(data[recordId]))
+          .includes(externalId)
+      );
+      const objectsThatExist = externalIds.filter(
+        (externalId) => !notFoundObjectExternalIds.includes(externalId)
+      );
+      return objectsThatExist;
+    }
+
+    // If unexpected error, re throw
```
Comment (freixedelo) [inline packages/ingestor/src/lib/skylark/saas/get.ts:97]: re throw?

File: packages/ingestor/src/lib/skylark/saas/graphql.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { GraphQLClient } from "graphql-request";
+import {
+  SAAS_ACCOUNT_ID,
+  SAAS_API_ENDPOINT,
+  SAAS_API_KEY,
+} from "../../constants";
+
+const account = new Date().toISOString();
+// const account = "test13"
```
Comment (freixedelo) [inline packages/ingestor/src/lib/skylark/saas/graphql.ts:9]: do you want to keep this ?

**Review by freixedelo — APPROVED** [review]

File: .github/actions/run-ingestor/action.yml (hunk 1 above)
Comment (james-wallis) [reply .github/actions/run-ingestor/action.yml:40]: Ye strings only

File: packages/ingestor/src/additional-objects/sets.ts (hunk 2 above)
Comment (james-wallis) [reply packages/ingestor/src/additional-objects/sets.ts:10]: is a note for future 😉

File: packages/ingestor/src/lib/skylark/saas/create.ts (hunk 3 above)
Comment (james-wallis) [reply packages/ingestor/src/lib/skylark/saas/create.ts:51]: good for debug

**Pull request merged**